### PR TITLE
[VMware/NGC]: Add Logs and ErrorHandlers for Storage Adapter

### LINKED ISSUE #272: VMware Storage Plugin: Add Logging and Improve Error Handling for Storage Adapter
**Is this a BUG REPORT or FEATURE REQUEST?**: 
/kind feature

**What happened**:
Need to add logging in Storage Adapters.
Add more Error Handling for the Adapters.

### PR DESCRIPTION
**Pull request by himanshuvar**
**What this PR does / why we need it**:
With reference to opensds/design-specs#38, This PR is to
support logging and Error handling for VMware vSphere OpenSDS storage plugin.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #272 

**Special notes for your reviewer**: 
This PR includes logging & Error Handling in Storage Adapters.

**Testing notes**:
1. Tested logging for Volume listing, creation for Storage Adapter.

### CODE REVIEW

File: vmware/ngc/NGC-AdapterManager/src/main/java/org/opensds/vmware/ngc/adapters/opensds/Main.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import org.opensds.vmware.ngc.models.VolumeMO;
+
+public class Main {
+	public static void main(String args[]) throws Exception {
+		OpenSDS openSDS = new OpenSDS("opensds");
+		openSDS.login("192.168.20.159", 50040, "admin", "opensds@123");
+		List<VolumeMO> list = openSDS.listVolumes();
+		//VolumeMO vol =openSDS.queryVolumeByID("688d5620cb544e49b8d759c0c517c16c1");
+		//System.out.println(vol.name);
+		for(VolumeMO volume : list) {
```
Comment (wisererik) [inline vmware/ngc/NGC-AdapterManager/src/main/java/org/opensds/vmware/ngc/adapters/opensds/Main.java]: remove unused code here

Comment (NajmudheenCT) [reply vmware/ngc/NGC-AdapterManager/src/main/java/org/opensds/vmware/ngc/adapters/opensds/Main.java]: Remove this file, hope this s for testing

**Comment by NajmudheenCT** [comment]
Remove this file

File: vmware/ngc/NGC-AdapterManager/src/main/java/org/opensds/vmware/ngc/adapters/opensds/Main.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package org.opensds.vmware.ngc.adapters.opensds;
```
Comment (kumarashit) [inline vmware/ngc/NGC-AdapterManager/src/main/java/org/opensds/vmware/ngc/adapters/opensds/Main.java]: How this file will work?

File: vmware/ngc/NGC-AdapterManager/src/main/java/org/opensds/vmware/ngc/adapters/opensds/RestClient.java [hunk 3]
```diff
@@ -132,29 +144,40 @@ void logout() {
...
+             logger.error(String.format("OpenSDS Logout Error: %s", e));
         } finally {
-            request = null;
+             request = null;
         }
     }
 
     void findDeviceInfo(String ip, int port) throws Exception {
-	    request.setUrl(String.format("http://%s:%d/v1beta", ip, port));
+        logger.info(String.format("OpenSDS Getting Storage Device info for OpenSDS Storage Device "
```
Comment (kumarashit) [inline vmware/ngc/NGC-AdapterManager/src/main/java/org/opensds/vmware/ngc/adapters/opensds/RestClient.java]: This message seems little recursive ;-)

File: vmware/ngc/NGC-AdapterManager/src/main/java/org/opensds/vmware/ngc/adapters/opensds/Constants.java [hunk 4]
```diff
@@ -75,14 +77,16 @@ public long getUnit()
...
+
 	private void init() {
 		if (properties == null) {
 			properties = new Properties();
 			try {
 				properties.load(this.getClass().getResourceAsStream(PATH));
 			}
 			catch (Exception ex) {
-				ex.printStackTrace();
+				logger.error(String.format("Error in loading Constant Properties Error Message is: %s", ex));
```
Comment (kumarashit) [inline vmware/ngc/NGC-AdapterManager/src/main/java/org/opensds/vmware/ngc/adapters/opensds/Constants.java]:
```suggestion
				logger.error(String.format("Error in loading Constant Properties, Error Message is: %s", ex));
```

File: vmware/ngc/NGC-AdapterManager/src/main/java/org/opensds/vmware/ngc/adapters/opensds/Main.java (hunk 1 above)
Comment (himanshuvar) [reply vmware/ngc/NGC-AdapterManager/src/main/java/org/opensds/vmware/ngc/adapters/opensds/Main.java]: Done, Removed this file

File: vmware/ngc/NGC-AdapterManager/src/main/java/org/opensds/vmware/ngc/adapters/opensds/Main.java (hunk 2 above)
Comment (himanshuvar) [reply vmware/ngc/NGC-AdapterManager/src/main/java/org/opensds/vmware/ngc/adapters/opensds/Main.java]: The file was for local testing purpose, pushed by mistake. 
I have removed this file.

File: vmware/ngc/NGC-AdapterManager/src/main/java/org/opensds/vmware/ngc/adapters/opensds/RestClient.java (hunk 3 above)
Comment (himanshuvar) [reply vmware/ngc/NGC-AdapterManager/src/main/java/org/opensds/vmware/ngc/adapters/opensds/RestClient.java]: Edited the message

File: vmware/ngc/NGC-AdapterManager/src/main/java/org/opensds/vmware/ngc/adapters/opensds/Constants.java (hunk 4 above)
Comment (himanshuvar) [reply vmware/ngc/NGC-AdapterManager/src/main/java/org/opensds/vmware/ngc/adapters/opensds/Constants.java]: Edited as suggested

**Pull request merged**